e2e/tests-dfx/extension.bash

```diff
@@ -115,3 +115,39 @@ echo testoutput' > "$CACHE_DIR"/extensions/test_extension/test_extension
     assert_match 'No extensions installed'
 }
 
+
+@test "run with hyphened parameters" {
+    CACHE_DIR=$(dfx cache show)
+    mkdir -p "$CACHE_DIR"/extensions/test_extension
+
+    # shellcheck disable=SC2016
+    echo '#!/usr/bin/env bash
+
+if [ "$2" == "--the-param" ]; then
+    echo "$3"
+fi' > "$CACHE_DIR"/extensions/test_extension/test_extension
+    chmod +x "$CACHE_DIR"/extensions/test_extension/test_extension
+    echo '{
+  "name": "test_extension",
+  "version": "0.1.0",
+  "homepage": "https://github.com/dfinity/dfx-extensions",
+  "authors": "DFINITY",
+  "summary": "Test extension for e2e purposes.",
+  "categories": [],
+  "keywords": [],
+  "subcommands": {
+    "abc": {
+      "about": "something something",
+      "args": {
+        "the_param": {
+          "about": "this is the param",
+          "long": "the-param"
+        }
+      }
+    }
+  }
+}' > "$CACHE_DIR"/extensions/test_extension/extension.json
+
+    assert_command dfx test_extension abc --the-param 123
+    assert_match "123"
```

Comment:
A couple things:
1. Match on a little bit more than the parameter value, to ensure that it's matching what we are looking for
2. Use "heredoc" for these inline file contents.  The quotes in `"EOF"` make it not perform variable substitution. 
```suggestion
    cat > "$CACHE_DIR"/extensions/test_extension/test_extension << "EOF"
#!/usr/bin/env bash

if [ "$2" == "--the-param" ]; then
    echo "the param is $3"
fi
EOF

    chmod +x "$CACHE_DIR"/extensions/test_extension/test_extension

    cat > "$CACHE_DIR"/extensions/test_extension/extension.json <<EOF
{
  "name": "test_extension",
  "version": "0.1.0",
  "homepage": "https://github.com/dfinity/dfx-extensions",
  "authors": "DFINITY",
  "summary": "Test extension for e2e purposes.",
  "categories": [],
  "keywords": [],
  "subcommands": {
    "abc": {
      "about": "something something",
      "args": {
        "the_param": {
          "about": "this is the param",
          "long": "the-param"
        }
      }
    }
  }
}
EOF

    assert_command dfx test_extension abc --the-param 123
    assert_contains "the param is 123"
```

Reply:
Oof, nice, I haven't thought about #1, and #2 is QoL improvement. Thanks!